agent/src/softwarecontaineragent.cpp

```diff
@@ -119,94 +120,113 @@ bool SoftwareContainerAgent::triggerPreload()
     return true;
 }
 
-inline bool SoftwareContainerAgent::isIdValid(ContainerID containerID)
+void SoftwareContainerAgent::assertContainerExists(ContainerID containerID)
 {
-    return (containerID < INT32_MAX)
-        && (containerID >= 0)
-        && (1 == m_containers.count(containerID));
+    if (containerID >= INT32_MAX) {
+        std::string errorMessage("Invalid Container ID: "
+                                + std::to_string(containerID)
+                                + ". ID can not be greater than INT32");
+        log_error() << errorMessage;
+        throw SoftwareContainerError(errorMessage);
+    }
+    
+    if (containerID < 0) {
+        std::string errorMessage("Invalid Container ID: "
+                                + std::to_string(containerID)
+                                + ". ID can not be negative");
+        log_error() << errorMessage;
+        throw SoftwareContainerError(errorMessage);
+    }
+    if (1 > m_containers.count(containerID)) {
```

Comment:
maybe it could be better to use i`f (!m_containers.count(containerID)) ` 
since it can be only 1 or 0, < operator doesn't seem relevant to me